client/src/featureform/serving.py

```diff
@@ -168,54 +166,70 @@ def process_transformation(self, name, variant):
         return new_data
 
     def _local_features(self, feature_variant_list, entity_tuple):
-        feature_dataframes = set()
-        dataframe_mapping = {}
+        if len(feature_variant_list) == 0:
+            raise Exception("No features provided")
+        dataframe_mapping = []
+        all_feature_df = None
         for featureVariantTuple in feature_variant_list:
 
             feature_row = self.sqldb.getNameVariant("feature_variant", "featureName", featureVariantTuple[0],
                                                     "variantName", featureVariantTuple[1])[0]
-            feature_column_name, source_name, source_variant = feature_row[11], feature_row[12], feature_row[13]
+            entity_column, ts_column, feature_column_name, source_name, source_variant = feature_row[9], feature_row[
+                10], feature_row[11], feature_row[12], feature_row[13]
             source_row = self.sqldb.getNameVariant("source_variant", "name", source_name, "variant", source_variant)[0]
             if self.sqldb.is_transformation(source_name, source_variant):
                 df = self.process_transformation(source_name, source_variant)
                 if isinstance(df, pd.Series):
                     df = df.to_frame()
-                #if feature_column_name not in df.columns:
                     df.reset_index(inplace=True)
                 df = df[[entity_tuple[0], feature_column_name]]
                 df.set_index(entity_tuple[0])
-
-                feature_dataframes.add(featureVariantTuple[0])
-                dataframe_mapping[featureVariantTuple[0]] = df
+                dataframe_mapping.append(df)
             else:
-                feature_dataframes, dataframe_mapping = self.process_feature_csv(source_row[10], entity_tuple[0],
-                                                                                 feature_dataframes,
-                                                                                 feature_column_name,
-                                                                                 dataframe_mapping,
-                                                                                 featureVariantTuple[0], "")
+                name_variant = "{}.{}".format(featureVariantTuple[0], featureVariantTuple[1])
```

Comment:
f-string is a cleaner way of doing formats as long as we only support Python >3.6. 
```
name_variant = "{featureVariantTuple[0]}.{featureVariantTuple[1]}"
```

Reply:
Done. I'm still in my Python2.7 days